### ISSUE TITLE
Collaboration Workflow

### ISSUE THREAD

**Issue by charleyburt** [charleyburt/Research_Fractals#8]
The workflow between the two of us should work as follows:

- You should be able to create issues, comment on issues, close issues, etc.
- If we want to work on the code we will:
  - create a new branch `$ git checkout -b your_branch_name`
  - commit and push on our branches (the very first time you push you will have to set the origin by typing `$ git push --set-upstream origin your_branch_name` but after that just `git push` will work)
  - Once we have a branch that is ready to merge with the `master` branch we will submit a `pull request` on GitHub (it can be done from command line but is somewhat difficult, and not as pretty, and kind of scary)
    - Just go to the [branches](https://github.com/charleyburt/Research_Fractals/branches) tab and click on `pull request`
      - From there we can have a discussion (i.e. "This is what I did and why I did it") in the `pull request` and then submit it, which will merge it with the `master`

**Comment by charleyburt** [comment]
Here is a test commit 

https://github.com/charleyburt/Research_Fractals/commit/4d3f421950e1d1526ef2dc1a7c7522c54aa8597a

**Comment by charleyburt** [comment]
Here is a test pull request (i.e. merging a branch into master)

https://github.com/charleyburt/Research_Fractals/pull/9#pullrequestreview-68392585

**Comment by charleyburt** [comment]
I think there are a lot of benefits that come with working through GitHub, especially the ability to keep track of `issues` and have discussion about them. We can reference websites https://en.wikipedia.org/wiki/Koch_snowflake or write some pseudo code:
```
this (is, some):
     test(code)
```
all of which are things we were previously doing through whiteboards, emails, google docs, etc.

I think having it all in one place will be advantageous for remembering what/why/when we did something. 

Also, I think it is cooler and good practice.

**Comment by kwilliams42** [comment]
Wow. I will try working on all of this tonight. I was trying to use GitHub Desktop or just in my editor (Atom), but will see if I need the command line.

**Comment by charleyburt** [comment]
Oh no, sorry, you shouldn't need the command line! Using GitHub Desktop you simply:

1) Click on Branch -> New Branch
2) Make changes using your text editor
3) Commit and Push whenever you want using the nice buttons
4) Click on Branch -> Create Pull Request when you are ready to merge into master!

**Comment by charleyburt** [comment]
Let me know if you have any questions, or if you think this isn't worth the hassle.
